lua/scratch/actor.lua

```diff
@@ -0,0 +1,137 @@
+local utils = require("scratch.utils")
+
+---@class Scratch.Actor
+---@field base_dir string
+---@field win_config vim.api.keyset.win_config @see: nvim_open_win() {config}
+---@field filetypes string[]
+---@field manual_text string
+---@field filetype_details Scratch.FiletypeDetails
+---@field localKeys Scratch.LocalKeyConfig[]
+local M = {}
+
+---@param filename string
+---@param win_conf? vim.api.keyset.win_config
+---@param content? string[]
+---@param local_keys? Scratch.LocalKeyConfig
+---@param cursor? Scratch.Cursor
+function M:scratchByName(filename, win_conf, content, local_keys, cursor)
```

Comment:
I think it makes sense to double this to something like `api.lua` or `utils.lua` in order to separate OOP and proc at every that looks func.